portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,104 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import com.google.sps.utilities.DatastoreHelper;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/my-form-handler")
+public class FormHandlerServlet extends HttpServlet {
+  private static final String TEXT = "comment-text";
+  private static final String NAME = "commenter-name";
+  private static final String EMAIL = "commenter-email";
+  private static final String IMAGE = "image";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, IMAGE);
+
+    // Get input from the form.
+    String text = getParameter(request, TEXT, "");
+    String commenterName = getParameter(request, NAME, "Anonymous");
+    String commenterEmail = getParameter(request, EMAIL, "Unknown");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(DatastoreHelper.buildCommentEntity(
```

Comment:
Just one long-term suggestion that since you use 5 params to build a COmmentEntity, you may want to use Builder for this.

You don't need to change it in this PR. Just something you may try to learn.

Reply:
Ack. Noted - will look into it, thank you.